tools/requirements.txt

```diff
@@ -1,5 +1,5 @@
-pygit2==1.4.0
-bidict==0.21.2
-Pillow==9.3.0
-PyYAML==5.4.1
-beautifulsoup4==4.10.0
+pygit2
+bidict
+Pillow
+PyYAML
+beautifulsoup4
```

Comment:
This is IMO a bad change. Not fixing dep versions can lead to broken setup, which can lead to mayhem. Instead, simply updating the dep versions would be gooder.

To do so:
remove the version tags
install in a venv
freeze again into the requirements.txt

Reply:
I could do that, but:

- It would make us depend on those versions, which could have some issues that get fixed when they are updated (pygit2 does)
- It creates technical debt, such that in N amount of time, some poor soul will have to go through all the dependencies again and update them all at once, instead of having a check fail (which would only happen if upstream breaks something anyways), see that it's one/two things to change, and either change them or simply push a new PR with a version binding until the fix (either upstream or on our codebase) is deployed
- Allows us to freely update the python version without having to worry about which version of which module is present or compatible with which release of python, pip will do it for us

(probably a question of preferences in the end, like many things in IT)

Reply:
I think fixed versions is better, so that we don't get caught by surprise when eventually we have to update.

Reply:
It is done